cpp/marc_tools/single_marc_convert_keywords.cc

```diff
@@ -0,0 +1,145 @@
+        ++record_count;
+
+        if (not record.getFirstSubfieldValue("150", 'a').empty())
+            gnd_keywords->emplace(std::make_pair(record.getFirstSubfieldValue("150", 'a'), record.getControlNumber()));
+        if (not record.getFirstSubfieldValue("150", 'g').empty())
+            gnd_keywords->emplace(std::make_pair(record.getFirstSubfieldValue("150", 'g'), record.getControlNumber()));
+        if (not record.getFirstSubfieldValue("150", 'x').empty())
+            gnd_keywords->emplace(std::make_pair(record.getFirstSubfieldValue("150", 'x'), record.getControlNumber()));
+    }
+
+    LOG_INFO("Processed " + std::to_string(record_count) + " MARC record(s).");
+    
+}
+
+
+void FindEquivalentKeywords(std::unordered_map<std::string, std::string> const *keywords_to_gnd, std::unordered_set<std::string> const *keywords_to_compare, const std::string matches_output_file, const std::string no_matches_output_file) {
+    std::unordered_map<std::string, std::string> keyword_matches;
+    std::unordered_set<std::string> keywords_without_match;
+    for (const auto &keyword : *keywords_to_compare) {
+        const auto lookup(keywords_to_gnd->find(keyword));
+        if (lookup != keywords_to_gnd->end()) {
+            keyword_matches.insert(*lookup);
+            continue;
+        }
+        keywords_without_match.insert(keyword);
+    }
+    LOG_INFO("Found " + std::to_string(keyword_matches.size()) + " keyword matches.\n");
+    double percentage = (static_cast<double>(keyword_matches.size())/static_cast<double>(keywords_to_compare->size())) * 100;
+    LOG_INFO("Which makes up for " + std::to_string(percentage) + "%\n");
+    LOG_INFO("Couldn't find a match for " + std::to_string(keywords_without_match.size()) + " keyword(s).\n");
+    std::ofstream output_file(matches_output_file);
+    for (const auto &[key, value] : keyword_matches) {
+        output_file << "Keyword:  " << key << " PPN: " << value << "\n";
+    }
+    std::ofstream out_file(no_matches_output_file);
+    for (const auto &word : keywords_without_match) {
+        out_file << word << "\n";
+    }
+}
+
+
+} // unnamed namespace
+
+
+void  ReadInKeywordsToCompare(const std::string &filename, std::unordered_set<std::string> * const keywords) {
+    std::ifstream input_file(filename.c_str());
+    if (input_file.fail())
+       throw std::runtime_error("in IniFile::processFile: can't open \"" + filename + "\"! ("
+                                 + std::string(::strerror(errno)) + ")");
+
+    // Read the file:
+    while (not input_file.eof()) {
+        std::string line;
+        // read lines until newline character is not preceeded by a '\'
+        bool continued_line;
+        do {
+            std::string buf;
+            std::getline(input_file, buf);
+            line += StringUtil::Trim(buf, " \t");
+            if (line.empty())
+                continue;
+
+            continued_line = line[line.length() - 1] == '\\';
```

Comment:
You don't need a separate "continued_line" variable.  Also, just use line.back().

Reply:
Please don't forget this change.